tests/torch/test_utils.py

```diff
@@ -83,3 +116,52 @@ def check_were_only_bn_training_state_changed(model: nn.Module, saved_state: _Mo
         check_were_only_bn_training_state_changed(model, saved_state)
 
     compare_saved_model_state_and_current_model_state(model, saved_state)
+
+
+@pytest.mark.parametrize(
+    "model",
+    [
+        BasicConvTestModel(),
+        TwoConvTestModel(),
+        MockModel(),
+        DepthWiseConvTestModel(),
+        EightConvTestModel(),
+        EmptyModel(),
+    ],
+)
+def test_model_device(model: nn.Module):
+    assert not is_multidevice(model)
+    assert get_model_device(model).type == "cpu"
+
+    multidevice = randomly_change_model_device(model)
+    assert is_multidevice(model) == multidevice
+
+    is_empty = is_model_empty(model)
+
+    if torch.cuda.is_available() and not is_empty:
+        cuda = torch.device("cuda")
+        model = model.to(cuda)
+        assert not is_multidevice(model)
+        assert get_model_device(model).type == "cuda"
```

Comment:
Please simplify the test:
1) Do not use `randomly_change_model_device`, test should be 100% deterministic
2) Please test only two models: empty model and Two conv test model as they cover 100% of cases
3) Please split this test on two tests: `test_empty_model_device` and `test_model_device` and remove this if section.

I expect following tests here:
`test_empty_model_device`: test creates empty model and check device
`test_model_device` skipped if cuda is not available, creates model on cpu, checks device and multidevice, move one param to cude, check devce and multidevce, move all params to cude, check device and multidevice